src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -313,64 +336,28 @@ static void freeKey(List<Section> sectionKey, ObjectPool<List<Section>> pool)
                 sectionKey.Clear();
                 pool.Free(sectionKey);
             }
+        }
 
-            static void addOptions(
-                AnalyzerConfig.Section section,
-                TreeOptions.Builder treeBuilder,
-                AnalyzerOptions.Builder analyzerBuilder,
-                ArrayBuilder<Diagnostic> diagnosticBuilder,
-                string analyzerConfigPath,
-                ConcurrentDictionary<ReadOnlyMemory<char>, string> diagIdCache)
+        internal TreeOptions MergeGlobalTreeOptions(TreeOptions options)
+        {
+            if (GlobalConfigOptions.TreeOptions?.IsEmpty == false)
```

Comment:
Unfortunate that the property is named as `TreeOptions`, it seems `DiagnosticOptions` is a much better name. I think this is a shipped public API that cannot be renamed, correct?